repoparse/analysis.py

```diff
@@ -0,0 +1,93 @@
+import json
+from sklearn.neighbors import KDTree
+from os import walk
+from typing import Dict, List
+
+
+def read_data(base_path: str = "./out", data_file_prefix: str = "repoparse_"):
+    merged_dict = {}
+    for root, dirs, files in walk(base_path):
+        for f in files:
+            if f.startswith(data_file_prefix):
+                with open(f, "r") as f_reader:
+                    repo_name = f[:len(data_file_prefix)].strip(".json")
+                    loaded_dict = json.load(f_reader)
+                    for _, author_dict in loaded_dict.items():
+                        for _, commit_dict in author_dict.items():
+                            # this is a mishap, please don't hit me too hard
+                            patch_set = commit_dict["patch_languages"]
+                            patch_languages = commit_dict["patch_set"]
+                            commit_dict["patch_languages"] = patch_languages
+                            commit_dict["patch_set"] = patch_set
+                    merged_dict[repo_name] = loaded_dict
+
+    return merged_dict
+
+
+def make_trigrams_cnt(identifier: str) -> Dict[str, int]:
+    res = {}
+    for trigram in map(lambda tup: ''.join(tup), zip(*[identifier[i:] for i in range(3)])):
+        if trigram not in res:
+            res[trigram] = 1
+        else:
+            res[trigram] += 1
+    return res
+
+# TODO generics
+def merge_cnts(left_dict, right_dict) -> dict:
```

Comment:
you can use [Counter](https://docs.python.org/3/library/collections.html#collections.Counter) from std lib